### PR TITLE
init bar chart MWPW-112993

### PR DESCRIPTION
**Pull request by meganthecoder**
* Initialize bar chart
* Add echarts lib
* Add chart light theme
* Add utils

Resolves: [MWPW-112993](https://jira.corp.adobe.com/browse/MWPW-112993)

**Test URLs:**
- Before: https://data-viz--milo--adobecom.hlx.page/drafts/methomas/data-viz/
- After: https://init-bar-chart--milo--adobecom.hlx.page/drafts/methomas/data-viz

### CODE REVIEW

**Review by Brandon32 — APPROVED** [review]

File: libs/blocks/chart/chart.js [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+    dataset: {
+      source: [
+        ['Browsers', 'Chrome', 'Firefox', 'Edge'],
+        ['Avg Visitors', 100, 156, 105],
+      ],
+    },
+    unit: 'k',
+  };
+
+  const colors = ['red', 'blue', 'green'];
```
Comment (JasonHowellSlavin) [inline libs/blocks/chart/chart.js:117]: Do you need a ToDo comment for this to replace with colors or are these the ones?

File: libs/blocks/chart/chart.js [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+  if (chartType !== 'oversizedNumber') {
+    console.log(chartType);
+    loadScript('/libs/deps/echarts.min.js')
+      .then(() => {
+        console.log('then');
```
Comment (JasonHowellSlavin) [inline libs/blocks/chart/chart.js:123]: console.log

File: libs/blocks/chart/chartLightTheme.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+  const MEDIUM = 'medium';
+  const SMALL = 'small';
+  const axisColor = '#767676';
+  let size;
+  let name;
+
+  if (deviceSize && (deviceSize === SMALL || deviceSize === MEDIUM)) {
+    size = SMALL;
+    name = 'consonantLightSmall';
+  } else {
```
Comment (JasonHowellSlavin) [inline libs/blocks/chart/chartLightTheme.js:26]: could remove the else and set the variables above to have large as a default. 
```
  let size = LARGE;
  let name = 'consonantLightLarge';

  if (deviceSize && (deviceSize === SMALL || deviceSize === MEDIUM)) {
    size = SMALL;
    name = 'consonantLightSmall';
  } 
```

File: test/blocks/chart/chart.test.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import { readFile } from '@web/test-runner-commands';
+import { expect } from '@esm-bundle/chai';
+
+const { getChartOptions } = await import('../../../libs/blocks/chart/chart.js');
+
+describe('chart', () => {
+  it('get bar chart options', async () => {
+    document.body.innerHTML = await readFile({ path: './mocks/bar.html' });
+    const options = getChartOptions('bar', {}, []);
+    expect(options).to.exist;
```
Comment (JasonHowellSlavin) [inline test/blocks/chart/chart.test.js:13]: Is the only way this tests fails is if options === undefined || false || null ? I believe, getChartOptions is always going to return an object, regardless of the params. 

There are some internal functionalities going on in getChartOptions, like deleting chartData, modifying and shifting seriesData. Would there be better ways to test these with dummy data, or would it require too much setup?

**Review by JasonHowellSlavin — COMMENTED** [review]
Great work! A few comments.

File: libs/blocks/chart/chart.js [hunk 5]
```diff
@@ -48,20 +56,111 @@ async function fetchData(link) {
...
+    legend: {
+      show: true,
+      type: 'scroll',
+    },
+    tooltip: {
+      show: true,
+      formatter: (params) => barTooltipFormatter(params, unit),
+    },
+    xAxis: {
+      type: chartType === 'bar' ? 'value' : 'category',
```
Comment (chrischrischris) [inline libs/blocks/chart/chart.js:99]: May want to create a `const isBarChart = chartType === 'bar'`
```suggestion
      type: isBarChart ? 'category' : 'value',
```

Comment (meganthecoder) [reply libs/blocks/chart/chart.js:99]: The conditional will likely change as we add more chart types, so I'd like to defer this until those are added

Comment (meganthecoder) [reply libs/blocks/chart/chart.js:99]: The conditional will likely change as we add more chart types, so I'd like to defer this until those are added

File: libs/blocks/chart/chart.js [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+  children[0]?.classList.add('chart_title');
+  children[1]?.classList.add('chart_subTitle');
+  children[3]?.classList.add('chart_footnote');
+
+  // ToDo: Replace hardcoded size MWPW-112994
+  const size = LARGE;
+  const chartRow = children[2];
+
+  if (chartRow) {
+    chartRow.style.width = '600px';
```
[the remark this answers is not in the document]
Comment (meganthecoder) [reply libs/blocks/chart/chart.js:129]: This is part of the TODO message a few lines up. It's only temporary

File: test/blocks/chart/chart.test.js (hunk 4 above)
Comment (meganthecoder) [reply test/blocks/chart/chart.test.js:13]: That sounds fine 👍

**Pull request merged**